synapseclient/models/dataset.py

```diff
@@ -2617,46 +2378,66 @@ async def main():
         if not any(current_item.id == item.id for current_item in self.items):
```

Comment:
This for loop does not account for the version, only the ID. If I am reading this right if `items` contains `syn1234` and version `1`, you wouldn't be able to add a second/more entries for version 1, 2, ect...